### PR TITLE
Violin plot (a kind of)

### PR DESCRIPTION
**Pull request by iliatimofeev**
Violinplot
---------

This example shows how to make a king of a Violinplot.

visualization - 2019-05-21T021104 068

### CODE REVIEW

**Comment by iliatimofeev** [comment]
with resolving of https://github.com/vega/vega-lite/issues/4939 I guess it could be simplified.

**Comment by jakevdp** [comment]
Should we wait for that update, or merge as-is?

File: altair/examples/violine_plot.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+This example shows how to make a king of a Violinplot.
+"""
+# category: other charts
+import altair as alt
+from vega_datasets import data
+
+source = data.cars.url
+
+violinplot =  alt.Chart(source).transform_filter(
+    alt.FieldGTPredicate(field='Miles_per_Gallon',gt=0)
```
Comment (jakevdp) [inline altair/examples/violine_plot.py]: I prefer using the form ``alt.datum.Miles_per_Gallon > 0``

File: altair/examples/violine_plot.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+).transform_bin(
+    ['bin_max', 'bin_min'], field='Miles_per_Gallon', bin=alt.Bin(maxbins=20)
+).transform_calculate(
+    binned='(datum["bin_max"]+datum["bin_min"])/2'
```
Comment (jakevdp) [inline altair/examples/violine_plot.py]: I prefer ``binned=(alt.datum.bin_max + alt.datum.bin_min) / 2``

Comment (iliatimofeev) [reply altair/examples/violine_plot.py]: done

**Comment by iliatimofeev** [comment]
>Should we wait for that update, or merge as-is?

Seems like density plots are quite requested (https://github.com/altair-viz/altair/issues/1494, https://github.com/altair-viz/altair/issues/1029) and with new impute transform we can somehow visualize density.

I don't how (and when) https://github.com/vega/vega-lite/issues/4939 will be resolved it may be that https://github.com/vega/vega-lite/issues/3442 will be resolved first. 

I opend https://github.com/altair-viz/altair/issues/1529 as reminder (if you assign it to me, I'll fix it)

**Pull request merged**